src/components/AssetInput/AssetInput.tsx

```diff
@@ -0,0 +1,230 @@
+                >
+                    {asset.type === "icon" ? (
+                        cloneElement(asset.icon, { size: IconSize.Size24 })
+                    ) : asset.type === "audio" ? (
+                        <IconAudio size={IconSize.Size24} />
+                    ) : (
+                        <img src={asset.src} alt="" className="tw-max-h-full" />
+                    )}
+                </span>
+                <span
+                    className={merge([
+                        size === "large" ? "tw-h-14" : "tw-h-full",
+                        "tw-py-3 tw-pr-3 tw-pl-4 tw-flex tw-flex-col tw-items-start tw-font-normal tw-flex-1",
+                    ])}
+                >
+                    <span
+                        className={merge([
+                            "tw-text-black-100 tw-text-s group-hover:tw-font-medium",
+                            (isOpen || isFocusVisible) && "tw-font-medium",
+                        ])}
+                    >
+                        {asset.name}
+                    </span>
+                    <span className="tw-flex tw-flex-row tw-items-center tw-gap-1 tw-text-black-80 tw-text-xxs">
+                        {asset.source === "library" ? (
+                            <>
+                                <IconImageLibrary />
+                                <span>{asset.sourceName}</span>
+                            </>
+                        ) : (
+                            <>
+                                <IconUploadAlternative />
+                                <span>Uploaded</span>
+                            </>
+                        )}
+                        {asset.extension && (
+                            <>
+                                <span className="tw-text-m tw-text-black-20 tw-h-4 tw-flex tw-items-center">•</span>
+                                <span>{asset.extension}</span>
+                            </>
+                        )}
+                        {asset.size && (
+                            <>
+                                <span className="tw-text-m tw-text-black-20 tw-h-4 tw-flex tw-items-center">•</span>
+                                <span>{asset.size}</span>
+                            </>
+                        )}
+                    </span>
+                </span>
+                <span className={merge([size === "large" ? "tw-h-14" : "tw-h-full", "tw-p-3 tw-flex tw-items-center"])}>
+                    <span className={merge(["tw-transition-transform", isOpen && "tw-rotate-180"])}>
+                        <IconCaretDown size={IconSize.Size16} />
+                    </span>
+                </span>
+            </button>
+
+            <AnimatePresence>
+                {isOpen && (
+                    <motion.div
+                        className="tw-absolute tw-left-0 tw-w-full tw-overflow-hidden tw-box-border tw-p-0 tw-shadow-mid tw-list-none tw-m-0 tw-mt-2"
+                        key="content"
```

Comment:
Consider adding a more unique `key` as it may lead to weird behaviour with framer's `AnimatePresence` if there are identical keys somewhere